pkg/route/api/v1/conversion.go

```diff
@@ -13,6 +13,9 @@ func init() {
 			obj.To.Kind = "Service"
 		},
 		func(obj *TLSConfig) {
+			if len(obj.Termination) == 0 && len(obj.DestinationCACertificate) == 0 {
+				obj.Termination = TLSTerminationEdge
```

Comment:
Yes, re-encrypt would require the destination ca certificate.  You could potentially not have the cert/ca cert on the route if you're terminating with a wildcard cert at the edge.